File: src/game/tools/rpc/output/wrap/GuildJoinRequestWrap.go
package wraps

import (
	rpc "tipcat.com/common/rpc/simple"
)

type GuildJoinRequestWrap struct {
	
	UID int64 // 玩家UID
	Name string // 名字
	Level int32 // 等级
	Score int32 // 积分
	Icon int32 // 头像
	Msg string // 留言
	rpc.Wrapper
}

func (w *GuildJoinRequestWrap)Decode(pck *rpc.Packet) rpc.Wrapper {
	
	w.UID = pck.PopInt64(); 
	w.Name = pck.PopString(); 
	w.Level = pck.PopInt32(); 
	w.Score = pck.PopInt32(); 
	w.Icon = pck.PopInt32(); 
	w.Msg = pck.PopString(); 
	return w
}

func (w *GuildJoinRequestWrap)Encode(pck *rpc.Packet) {
	
	pck.PutInt64(w.UID) 
	pck.PutString(w.Name) 
	pck.PutInt32(w.Level) 
	pck.PutInt32(w.Score) 
	pck.PutInt32(w.Icon) 
	pck.PutString(w.Msg) 
}

File: src/game/tools/rpc/output/wrap/GuildJoinRequestWrap.cs
using System;
using Common.Net.Simple;

namespace Game 
{
	public class GuildJoinRequestWrap : IWrapper
	{	
		public long UID; // 玩家UID
		public string Name; // 名字
		public int Level; // 等级
		public int Score; // 积分
		public int Icon; // 头像
		public string Msg; // 留言
		
		public void Decode(Packet pck)
		{	
			UID = pck.GetLong(); 
			Name = pck.GetString(); 
			Level = pck.GetInt(); 
			Score = pck.GetInt(); 
			Icon = pck.GetInt(); 
			Msg = pck.GetString(); 
		}
		
        public void Encode(Packet pck)
        {	
        	pck.PutLong(UID); 
        	pck.PutString(Name); 
        	pck.PutInt(Level); 
        	pck.PutInt(Score); 
        	pck.PutInt(Icon); 
        	pck.PutString(Msg); 
        }
	}
}